### ISSUE TITLE
`TRQ_MODE` overshooting.

### ISSUE THREAD

**Issue by Blue186** [EFeru/hoverboard-firmware-hack-FOC#455]
### Variant

USART

### Control type

FOC

### Control mode

None

### Description

Hello, I have been trying to implement control mode switching through serial messages using SERIAL_START_FRAME. I observed that when in USART+FOC_CTRL+SPD_MODE/VLT_MODE, everything works fine, and the mode switch is successful. However, when attempting USART+FOC_CTRL+SPD_MODE/TRQ_MODE, after switching to TRQ_MODE, there are no errors, but when I slightly change command_in->steer or command_in->speed, the motor speed increases directly to the maximum.When I use the `TRQ_MODE` mode alone, the same issue occurs when combined with serial communication. I have tried reading through the code but haven't identified the issue. I would greatly appreciate your help if you could assist me in resolving this problem.
- 
![image](https://github.com/EFeru/hoverboard-firmware-hack-FOC/assets/56758678/1e5f899b-0ade-44d0-87e5-e101e1122793)
![image](https://github.com/EFeru/hoverboard-firmware-hack-FOC/assets/56758678/062e7aa6-3cdd-4e69-ba4b-cd9d832ec5a6)

**Comment by Candas1** [comment]
Hi,

It's hard to follow you.
If it's happening with torque mode only, why share all the unrelated details about changing the mode?

This is how torque mode works without load. 

Changing the mode is already covered with a different protocol if you read the wiki.

**Comment by Blue186** [comment]
Thanks a lot, yes I found it. I found through issues that the torque mode is not linear and not very adjustable, so can I control the input of the torque mode through PID? Calculate the speed difference of the controlled variable, then I can control the speed in the torque mode, so as to better adjust the torque? (Not FOC speed mode) I will try that later.

**Comment by Candas1** [comment]
[This](https://github.com/Candas1/Sideboard-Arduino/blob/master/src/Hoverserial.cpp) project is unfinished but you might be able to use the protocol piece.

For the non linearity, I think it's more a scale problem, it's mentioned on those issues.

**Comment by Blue186** [comment]
Ok,thanks!
I'll try to adapt it.
In addition, I would like to ask, when I use a motor with a reduction ratio of 1:4.4 and 10 pole pairs, everything is normal when the motor rotates in the forward direction, but when it rotates in the reverse direction, it will stall and cannot start. ,what could be the problem? Of course, I also modified n_polePairs: 10 in the BLDC_controller_data.c file, and also tried to increase r_errInpTgtThres and t_errQual.

**Comment by Blue186** [comment]
> Ok,thanks! I'll try to adapt it. In addition, I would like to ask, when I use a motor with a reduction ratio of 1:4.4 and 10 pole pairs, everything is normal when the motor rotates in the forward direction, but when it rotates in the reverse direction, it will stall and cannot start. ,what could be the problem? Of course, I also modified n_polePairs: 10 in the BLDC_controller_data.c file, and also tried to increase r_errInpTgtThres and t_errQual.

This happens from time to time.

**Comment by Candas1** [comment]
Are the phases/halls correct ?

**Comment by lebedyncrs** [comment]
> Hi,
> 
> It's hard to follow you. If it's happening with torque mode only, why share all the unrelated details about changing the mode?
> 
> This is how torque mode works without load.
> 
> Changing the mode is already covered with a different protocol if you read the wiki.

Can you please share wiki about changing mode? Cannot find it

**Comment by Blue186** [comment]
> > 你好，
> > 很难理解你的意思。如果这种情况只发生在扭矩模式下，为什么要分享有关更改模式的所有不相关的细节？
> > 这就是扭矩模式无负载的工作方式。
> > 如果你读过 wiki，就会发现更改模式已经包含在不同的协议中。
> 
> 你能分享关于更改模式的 wiki 吗？找不到

hello,
The torque mode is indeed not linear. If you need to modify the control mode and parameters first, you can do so in the config file, as shown in the following figure
![image](https://github.com/user-attachments/assets/dafd26ba-257e-4898-a81f-2f991f5f2c93)
